shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -309,12 +309,18 @@ def name(self) -> str:
         return self.fx_node.name
 
     @property
-    def node_args(self) -> list[Any]:
+    def node_args(self) -> dict[int, Any]:
         """Returns the args to this custom op using subclasses of CustomOp if possible."""
-        return [
-            get_custom(arg) if isinstance(arg, fx.Node) else arg
-            for arg in self.fx_node.args
-        ]
+        custom_args = {}
+        for i, arg in enumerate(self.fx_node.args):
+            if isinstance(arg, fx.Node):
+                custom_args[i] = get_custom(arg)
+            if isinstance(arg, list) and all(isinstance(x, fx.Node) for x in arg):
+                custom_args[i] = [get_custom(x) for x in arg]
+        return custom_args
+
+    def node_arg_index(self, arg: CustomOp) -> Optional[CustomOp | list[CustomOp]]:
```

Comment:
NIT: I think would be more inutiiative to call it `get_arg_index` or `get_operand_index` or something along that line.